http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -256,6 +256,46 @@ class HttpSamDAO[F[_]](httpClient: Client[F],
         }
     } yield ()
 
+  def createResourceWithWorkspaceParent[R](resource: R,
+                                           creatorEmail: WorkbenchEmail,
+                                           userInfo: UserInfo,
+                                           workspaceId: WorkspaceId
+  )(implicit
+    sr: SamResource[R],
+    encoder: Encoder[R],
+    ev: Ask[F, TraceId]
+  ): F[Unit] =
+    for {
+      traceId <- ev.ask
+      // TODO: Check if this is the best way to get the token, there are no petSA for Workspace resources, only google-projects
+      authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
```

Comment:
I went down a rabbit hole in the SAM API documentation, but could not find a better way to get the token, am I missing something?

Reply:
Fine as long as called synchronously, but maybe use this instead `getCachedArbitraryPetAccessToken`

Reply:
Do this to stay consistent with the Azure resources

Reply:
Thinking more, I feel like using `UserInfo` is fine. The other branch in `SamAuthProvider.notifyResourceCreatedV2` calls `createResourceWithUserInfo` so we're already using `UserInfo` for runtime creations. Doing similar for apps would avoid an extra Sam hop. If we ever need to do this asynchronously we can revisit.

Reply:
Sounds good, I will leave a comment in the code to explain